lib/uksched/include/uk/sched.h

```diff
@@ -98,6 +98,11 @@ struct uk_sched {
 	struct uk_alloc *a_auxstack; /**< default allocator for aux stacks */
 	struct uk_alloc *a_uktls; /**< default allocator for TLS+ectx */
 	struct uk_sched *next;
+#if CONFIG_LIBUKSCHED_STATS
+	unsigned long num_yields;
+	unsigned long num_schedules;
+	unsigned long num_switches;
```

Comment:
I think those would make more sense to have per thread?
How are they also different between each other?
I remember Linux had stats like `number of voluntary context switches` (sched_yield) and `number of involuntary context switches` (could be an expired time slice).

Reply:
We can definitely do that. What we implement in this PR is CPU aggregate statistics, i.e. the equivalent of `/proc/schedstat`. Per-thread stats involve a few more challenges:

* Application threads need to be provided by `posix-process`, when that is enabled. We can store ukstore objects in the thread struct that `posix-process` can access when constructing the process hierarchy. FTR, linux provides aggregate process stats `/proc/<pid>/schedstat` as well as per-thread stats `/proc/<pid>/task/<tid>/schedstat`.
* Kernel threads need some sort of identifier in lieu of TID, which the caller must provide.

Given the above, I see this as a separate task, so I propose that we handle this on a separate PR.

Reply:
Sounds good, I agree.